examples/asynchronous_stopping_criterion/asynchronous_stopping_criterion.cpp

```diff
@@ -129,12 +129,13 @@ class ByInteraction : public EnablePolymorphicObject<ByInteraction, Criterion> {
 void run_solver(volatile bool *stop_iteration_process,
                 std::shared_ptr<gko::Executor> exec)
 {
+    // Some shortcuts
     using mtx = gko::matrix::Csr<>;
     using vec = gko::matrix::Dense<>;
     using bicg = gko::solver::Bicgstab<>;
 
     // Read Data
-    auto A = gko::share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
+    auto A = share(gko::read<mtx>(std::ifstream("data/A.mtx"), exec));
```

Comment:
How come that works, we're not in the `gko` namespace right?

Reply:
I also though it wouldn't work for a while, but it's good enough for ADL to kick in. From [cppreference](https://en.cppreference.com/w/cpp/language/adl):

> 3) For arguments whose type is a class template specialization, in addition to the class rules, the following types are examined and their associated classes and namespaces are added to the set
>     a) The types of all template arguments provided for type template parameters (skipping non-type template parameters and skipping template template parameters)

So, since `exec` has the type `std::shared_ptr<gko::Executor>`, the `gko::Executor` template argument adds the `gko` namespace to the list of namespaces ADL is considering.

Reply:
Actually, no, slightly wrong explanation, but the same reason. It's the result of `gko::read<mtx>`, which is `std::unique_ptr<gko::matrix::Csr>>`, that adds `gko` and `gko::matrix` to ADL's list.

Reply:
Ah I see. Though that means that unlike before, we may have multiple choice for those functions. Is it a good idea to show users that? What if they have their own `share` laying somewhere in their code?